crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -228,12 +265,37 @@ impl QueryLimitsChecker {
                                 f.pos,
                             ));
                         }
+
+                        let (current_count, is_connection) =
+                            estimate_output_nodes_for_curr_node(f, variables, parent_is_connection)
+                                .map_err(|_| {
+                                    graphql_error_at_pos(
+                                        BAD_USER_INPUT,
```

Comment:
Is this a BAD_USER_INPUT kind of error?  Looking at the message, it indicates to me that it's something else.

Reply:
Yes, the possible errors are all client fault. For example, using a variable but not providing a value for the variable, passing in a non-number value, etc. Maybe BAD_REQUEST is a better flag?